test/config-message-function.js

```diff
@@ -0,0 +1,462 @@
+'use strict';
+
+var expect = require('expect');
+var exec = require('child_process').exec;
+var path = require('path');
+var fs = require('fs');
+var os = require('os');
+
+var tildify = require('../lib/shared/tildify');
+
+var baseDir = path.join(__dirname, 'fixtures/config/theming');
+var expectedDir = path.join(__dirname, 'expected/config/theming');
+
+var eraseTime = require('./tool/erase-time');
+var eraseLapse = require('./tool/erase-lapse');
+var sliceLines = require('./tool/slice-lines');
+var gulp = require('./tool/gulp-cmd');
+
+describe('config: message function', function() {
+
+  it('can change USAGE with .gulp.*', function(done) {
+    var cwd = path.join(baseDir, 'USAGE');
+    var expected = fs.readFileSync(path.join(expectedDir, 'usage.txt'), 'utf8');
+
+    var opts = { cwd: cwd };
+    exec(gulp('--help'), opts, cb);
+
+    function cb(err, stdout, stderr) {
+      expect(err).toBeNull();
+      expect(stderr).toEqual('');
+      expect(stdout).toEqual(expected);
+      done(err);
+    }
+  });
+
+  it('can change flag descriptions with .gulp.*', function(done) {
+    var cwd = path.join(baseDir, 'flags');
+    var expected = fs.readFileSync(path.join(expectedDir, 'flags.txt'), 'utf8');
+
+    var opts = { cwd: cwd };
+    exec(gulp('--help'), opts, cb);
+
+    function cb(err, stdout, stderr) {
+      expect(err).toBeNull();
+      expect(stderr).toEqual('');
+      expect(stdout).toEqual(expected);
+      done(err);
+    }
+  });
+
+  it('can change DESCRIPTION with .gulp.*', function(done) {
+    var cwd = path.join(baseDir, 'DESCRIPTION');
+    var expected = '**DESCRIPTION**\n' +
+      '└── default\n';
+
+    var opts = { cwd: cwd };
+    exec(gulp('--tasks'), opts, cb);
+
+    function cb(err, stdout, stderr) {
+      expect(err).toBeNull();
+      expect(stderr).toEqual('');
+      expect(stdout).toEqual(expected);
+      done(err);
+    }
+  });
+
+  // TODO: Do we want tests returning false for every message?
```

Comment:
@sttk what do you think about this?

Reply:
If `message` function always a line, and outputting the line is controlled in common process, the test is sufficient for just one message, I think.